File: frontend/src/types/canva.ts
import type { GeneratedContent } from "./content";

export type CanvaGraphicType = "single" | "carousel" | "quote";

export type CanvaGeneratedCopy = {
  headline: string;
  bodyText: string;
  callToAction: string;
};

export type CanvaGraphicRequest = {
  graphicType: CanvaGraphicType;
  content: GeneratedContent;
};

export type CanvaDesignResult = {
  url?: string;
  thumbnailUrl?: string;
  jobId?: string;
};

export type CanvaGraphicResponse = {
  status: "created" | "in_progress" | "setup_required";
  graphicType: CanvaGraphicType;
  extractedContent: CanvaGeneratedCopy;
  message: string;
  design?: CanvaDesignResult;
};

File: backend/src/invictus_os/schemas/canva.py
from typing import Literal

from pydantic import BaseModel, Field

from invictus_os.schemas.content import GeneratedContentResponse

CanvaGraphicType = Literal["single", "carousel", "quote"]
CanvaDesignStatus = Literal["created", "in_progress", "setup_required"]


class CanvaGeneratedCopy(BaseModel):
    headline: str = Field(min_length=1)
    body_text: str = Field(min_length=1)
    call_to_action: str = Field(min_length=1)


class CanvaGraphicRequest(BaseModel):
    graphic_type: CanvaGraphicType
    content: GeneratedContentResponse


class CanvaDesignResult(BaseModel):
    url: str | None = None
    thumbnail_url: str | None = None
    job_id: str | None = None


class CanvaGraphicResponse(BaseModel):
    status: CanvaDesignStatus
    graphic_type: CanvaGraphicType
    extracted_content: CanvaGeneratedCopy
    message: str
    design: CanvaDesignResult | None = None
